etna/pipeline/base.py

```diff
@@ -1,4 +1,5 @@
 import math
+import reprlib
```

Comment:
Why do we actually have so many changes? What the difference between "int" and "timestamp"?

Reply:
There are so many changes because `FoldMask` wasn't tested in isolation, it was only tested in backtest methods. So, I had to add tests for this class.

Reply:
No, I mean in the file with FoldMask it self

Reply:
- Docstrings were updated.
- New validations were added to take into account possible parameters of mismatched types.
- I changed `validate_on_dataset` method by replacing some validations with others. It should be discussed, but I thought new validations are more clear.